packages/common/src/hardhat/tasks/xchainSetup.ts

```diff
@@ -17,11 +20,103 @@ async function setupHub(hub: Contract, deployer: string, parentMessenger: string
   }
 }
 
+async function setupParentMessenger(
+  messenger: Contract,
+  deployer: string,
+  childMessenger: Deployment,
+  oracleHub: Deployment,
+  governorHub: Deployment,
+  oracleSpoke: Deployment,
+  governorSpoke: Deployment
+) {
+  const contractState = await Promise.all([
+    messenger.methods.getL1CallValue().call(),
+    messenger.methods.owner().call(),
+    messenger.methods.childMessenger().call(),
+    messenger.methods.oracleHub().call(),
+    messenger.methods.oracleSpoke().call(),
+    messenger.methods.governorHub().call(),
+    messenger.methods.governorSpoke().call(),
+  ]);
+  const messengerOwner = contractState[1];
+  const messengerChildMessenger = contractState[2];
+  const messengerOracleHub = contractState[3];
+  const messengerOracleSpoke = contractState[4];
+  const messengerGovernorHub = contractState[5];
+  const messengerGovernorSpoke = contractState[6];
```

Comment:
you can do this cleaner using different syntax
```suggestion
const [
  messengerOwner,
  messengerChildMessenger,
  messengerOracleHub,
  messengerOracleSpoke,
  messengerGovernorHub,
  messengerGovernorSpoke,
] = await Promise.all([
  messenger.methods.getL1CallValue().call(),
  messenger.methods.owner().call(),
  messenger.methods.childMessenger().call(),
  messenger.methods.oracleHub().call(),
  messenger.methods.oracleSpoke().call(),
  messenger.methods.governorHub().call(),
  messenger.methods.governorSpoke().call(),
]);
```

Reply:
also does not look like you use `messenger.methods.getL1CallValue`

Reply:
But this doesn't work because `getL1CallValue` isn't used, but we need it in the calling function to determine how much ETH to send to messenger

Reply:
you could just set it to a value in the assignment array then not use that? meh, either implementation is fine :)